flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneJobClusterEntryPoint.java

```diff
@@ -77,7 +81,9 @@ private StandaloneJobClusterEntryPoint(
 	protected DispatcherResourceManagerComponentFactory<?> createDispatcherResourceManagerComponentFactory(Configuration configuration) {
 		return new JobDispatcherResourceManagerComponentFactory(
 			StandaloneResourceManagerFactory.INSTANCE,
-			new ClassPathJobGraphRetriever(jobId, savepointRestoreSettings, programArguments, jobClassName));
+			new ClassPathJobGraphRetriever(jobId, savepointRestoreSettings, programArguments, jobClassName,
+				Paths.get(System.getenv(ENV_FLINK_HOME_DIR) == null ? DEFAULT_FLINK_HOME : System.getenv(ENV_FLINK_HOME_DIR),
```

Comment:
Hard coding "/opt/flink" looks a bit tricky. 
Is it possible to we set "FLINK_HOME" in all cases? So that we can throw exception here for unexpected cases.